planet/cli/data.py

```diff
@@ -480,8 +483,139 @@ async def search_update(ctx,
         echo_json(items, pretty)
 
 
+@data.command()
+@click.pass_context
+@translate_exceptions
+@coro
+@click.argument("item_type",
+                type=types.CommaSeparatedString(),
+                callback=check_item_types)
+@click.argument("item_id")
+@click.argument("asset_type_id")
+@click.option('--directory',
+              default='.',
+              help=('Base directory for file download.'),
+              type=click.Path(exists=True,
+                              resolve_path=True,
+                              writable=True,
+                              file_okay=False))
+@click.option('--filename',
+              default=None,
+              help=('Custom name to assign to downloaded file.'),
+              type=str)
+@click.option('--overwrite',
+              is_flag=True,
+              default=False,
+              help=('Overwrite files if they already exist.'))
+@click.option('--checksum',
+              is_flag=True,
+              default=None,
+              help=('Verify that checksums match.'))
+async def asset_download(ctx,
+                         item_type,
+                         item_id,
+                         asset_type_id,
+                         directory,
+                         filename,
+                         overwrite,
+                         checksum):
+    """Download an activated asset.
+
+    This function will fail if the asset state is not activated. Consider
+    calling `asset-wait` before this command to ensure the asset is activated.
+
+    If --checksum is provided, the associated checksums given in the manifest
+    are compared against the downloaded files to verify that they match.
+
+    If --checksum is provided, files are already downloaded, and --overwrite is
+    not specified, this will simply validate the checksums of the files against
+    the manifest.
+
+    Output:
+    The full path of the downloaded file. If the quiet flag is not set, this
+    also provides ANSI download status reporting.
+    """
+    quiet = ctx.obj['QUIET']
+    async with data_client(ctx) as cl:
+        asset = await cl.get_asset(item_type.pop(), item_id, asset_type_id)
+        path = await cl.download_asset(asset=asset,
+                                       filename=filename,
+                                       directory=Path(directory),
+                                       overwrite=overwrite,
+                                       progress_bar=not quiet)
+        if checksum:
+            cl.validate_checksum(asset, path)
+
+
+@data.command()
+@click.pass_context
+@translate_exceptions
+@coro
+@click.argument("item_type",
```

Comment:
I agree about the string, but are you saying we don't want to validate `item_type` with the callback function? The response from the server for an unsupported `item_type` is not very helpful.

Reply:
However, handing off to the callback function with `item_type` as a string does cause issues, because it cycles through `item_type in item_types`, so handing off something like `"PSScene"` raises a `click.BadParameter()` because "P" isn't a valid `item_type`.

Reply:
Went with option 2, see: https://github.com/planetlabs/planet-client-python/pull/864/commits/f01a519a1addcbd2ddd8e2ee64ab1fa33a168647